OpenSim/Common/Logger.cpp

```diff
@@ -150,6 +150,16 @@ void Logger::addFileSink(const std::string& filepath) {
              "removeFileSink() first.", m_filesink->filename());
         return;
     }
+    // verify if file can be opened at the specified path
+    std::ifstream ifs(filepath);
+
+    if (ifs.is_open()) {
+        ifs.close();
+    } else {
+        // show message:
+        std::cout << "Can't open file " << filepath << "Log will not be created." << std::endl;
```

Comment:
Should use the logger: `log_warn(...)`.

We could add a function like `Logger::fileSinkExists()` or `Logger::hasFileSink()` that the GUI could invoke to determine if the GUI should log to some user data directory.

I think it's also fine for there to be no logging file when using the GUI—the content is in the Messages window, anyway.

Reply:
@chrisdembia The function addFileSink() is invoked from inside the constructor of Logger so it's not clear if any other logging is possible at this point, including log_cout or log_warn 😕 

I think we can make the GUI add its own FileSink as needed but no logging from GUI for now (if the file can't be created) seems less catastrophic than the current behavior.

Reply:
That's a good point but `addFileSink()` could be invoked by a user after the constructor has completed, in which case we would want `log_warn()` to be used.